### PR TITLE
hotfix/extensions-500-fix

### PR DESCRIPTION
**Pull request by shayanaijaz**
## Overview

Fix for 500 error on View Extensions page

## Related

…

## Changes

1- Since `applicable_data_period` field is stored as an `int` in the database and not `datetime` I changed how the conversion to a readable date value was happening

## Testing

1. In order to mock some data for local testing make changes to the following files

`apcd_cms/src/apps/admin_extensions/views.py:`

<details>
       <summary>View Snippet</summary>

Add the following to the top of the file:

`import datetime`

 Replace line 37 with the following :

  ```js
extension_content = [
        (
            1,                                                                      # extension_id
            1,                                                                      # submitter_id
            datetime.date(2023, 12, 31),                                            # current_expected_date
            datetime.date(2023, 12, 31),                                            # requested_target_date
            None,                                                                   # approved_expiration_date
            'other',                                                                # extension_type
            None,                                                                   # applicable_data_period
            None,                                                                   # status
            'pending',                                                              # outcome
            datetime.datetime(2022, 12, 20, 11, 10, 34, 598953),                    # created_at
            None,                                                                   # updated_at
            'TESTGOLD',                                                             # submitter_code
            10000000,                                                               # payor_code
            'jharri66',                                                             # user_id
            'Joe Harrison',                                                         # requestor_name
            'joseph.harrison@uth.tmc.edu',                                          # requestor_email
            'Currently, our claims process does not require a street address.',     # explanation_justification
            None,                                                                   # notes
            'CHCD'                                                                  # org_name
        ),
        (
            2,                                                                      # extension_id
            2,                                                                      # submitter_id
            datetime.date(2023, 12, 31),                                            # current_expected_date
            datetime.date(2023, 12, 31),                                            # requested_target_date
            None,                                                                   # approved_expiration_date
            'other',                                                                # extension_type
            202308,                                                                 # applicable_data_period
            None,                                                                   # status
            'pending',                                                              # outcome
            datetime.datetime(2022, 12, 20, 11, 10, 34, 598953),                    # created_at
            None,                                                                   # updated_at
            'TESTGOLD',                                                             # submitter_code
            10000000,                                                               # payor_code
            'nguyken',                                                              # user_id
            'Kenneth Nguyen',                                                       # requestor_name
            'kenneth.nguyen@uth.tmc.edu',                                           # requestor_email
            'Currently, our claims process does not require a street address.',     # explanation_justification
            None,                                                                   # notes
            'CHCD'                                                                  # org_name
        ),
        (
            3,                                                                      # extension_id
            1,                                                                      # submitter_id
            datetime.date(2023, 12, 31),                                            # current_expected_date
            datetime.date(2023, 12, 31),                                            # requested_target_date
            None,                                                                   # approved_expiration_date
            'other',                                                                # extension_type
            202308,                                                                 # applicable_data_period
            None,                                                                   # status
            'pending',                                                              # outcome
            datetime.datetime(2022, 12, 20, 11, 10, 34, 598953),                    # created_at
            None,                                                                   # updated_at
            'TESTGOLD',                                                             # submitter_code
            10000000,                                                               # payor_code
            'nguyken',                                                              # user_id
            'Kenneth Nguyen',                                                       # requestor_name
            'kenneth.nguyen@uth.tmc.edu',                                           # requestor_email
            'Currently, our claims process does not require a street address.',     # explanation_justification
            None,                                                                   # notes
            'CHCD'                                                                  # org_name
        ),
        (
            4,                                                                      # extension_id
            58,                                                                     # submitter_id
            datetime.date(2023, 12, 31),                                            # current_expected_date
            datetime.date(2023, 12, 31),                                            # requested_target_date
            None,                                                                   # approved_expiration_date
            'other',                                                                # extension_type
            202308,                                                                 # applicable_data_period
            None,                                                                   # status
            'pending',                                                              # outcome
            datetime.datetime(2022, 12, 20, 11, 10, 34, 598953),                    # created_at
            None,                                                                   # updated_at
            'TESTGOLD',                                                             # submitter_code
            10000000,                                                               # payor_code
            'dias',                                                                 # user_id
            'Devon Miller',                                                         # requestor_name
            'dias@lfg.com',                                                         # requestor_email
            'Currently, our claims process does not require a street address.',     # explanation_justification
            None,                                                                   # notes
            'CHCD'                                                                  # org_name
        )
    ]
  ```

</details>

2. Open [http://localhost:8000/administration/list-extensions](http://localhost:8000/administration/list-extensions)
4. Ensure page loads fine and the `applicable_data_period` field in the detail modal shows either None or the correct month and year

## UI
Screen Shot 2023-02-09 at 4 29 05 PM

### CODE REVIEW

**Review by sophia-massie — APPROVED** [review]
LGTM

File: apcd-cms/src/apps/admin_extension/views.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def _set_extensions(extension):
                 'requested_target_date': extension[3],
                 'approved_expiration_date': extension[4],
                 'extension_type': title_case(extension[5]),
-                'applicable_data_period': extension[6].date() if extension[6] else None,
+                'applicable_data_period': datetime.datetime.strptime(str(extension[6]), '%Y%m').strftime('%b. %Y') if extension[6] else None,
```
Comment (fnets) [inline apcd-cms/src/apps/admin_extension/views.py:47]: There's a lot of typecasting going on right here. Is there any way to simplify it? Why is all of this required for this field but not the other date fields?

**Review by fnets — CHANGES_REQUESTED** [review]

File: apcd-cms/src/apps/admin_extension/views.py (hunk 1 above)
Comment (shayanaijaz) [reply apcd-cms/src/apps/admin_extension/views.py:47]: This field is the only date field that needs to be stored as an int in the form YYYYMM (e.g 202302). Not sure why that is but that's how the table is set up. I'm open to any alternative ways to do this as well

**Review by sophia-massie — APPROVED** [review]
LGTM

**Review by fnets — APPROVED** [review]
LGTM

**Pull request merged**